Backend/Remora.Discord.Rest/DiscordHttpClient.cs

```diff
@@ -68,7 +68,7 @@ IOptions<JsonSerializerOptions> serializerOptions
         /// </summary>
         /// <param name="requestCustomizer">The action that customizes the request.</param>
         /// <returns>The created customization.</returns>
-        public DiscordRequestCustomization AddCustomization(Action<RestRequestBuilder> requestCustomizer)
+        public DiscordRequestCustomization WithCustomization(Action<RestRequestBuilder> requestCustomizer)
```

Comment:
So, wait, after an hour's worth of arguing, this ALREADY EXISTS?!

Why the name change here?

Reply:
I thought I made that clear, and that it was because of its existence that I originally wanted to expose `DiscordHttpClient` as the main modification point.

The name change is just personal preference. I think `WithCustomization` works better than `AddCustomization`, but it's either or.

Reply:
I dunno, that must've been before I joined in the convo.

I also like that name better, and I just realized it's not a breaking change, cause that addition hasn't been published yet, so nevermind.